src/rust/src/x509/certificate.rs

```diff
@@ -207,18 +207,26 @@ impl Certificate {
                         ),
                     ));
                 }
-                let filtered_extensions: Extensions<'_> = Asn1ReadableOrWritable::new_write(
+                let filtered_extensions: RawExtensions<'_> = Asn1ReadableOrWritable::new_write(
                     asn1::SequenceOfWriter::new(filtered_extensions),
                 );
-                tbs_precert.extensions = Some(filtered_extensions);
+                tbs_precert.raw_extensions = Some(filtered_extensions);
                 let result = asn1::write_single(&tbs_precert)?;
                 Ok(pyo3::types::PyBytes::new(py, &result))
             }
-            None => Err(CryptographyError::from(
+            Ok(None) => Err(CryptographyError::from(
                 pyo3::exceptions::PyValueError::new_err(
                     "Could not find any extensions in TBS certificate",
                 ),
             )),
+            Err(oid) => {
+                let oid_obj = oid_to_py_oid(py, &oid)?;
+                Err(exceptions::DuplicateExtension::new_err((
+                    format!("Duplicate {} extension found", oid),
+                    oid_obj.into_py(py),
+                ))
+                .into())
+            }
```

Comment:
To avoid needing to repeat this stanza anywhere we touch extensions, can we make `extensions` return a more-specific error type and then we can do a `From` impl on that?

Reply:
Yeah, good idea -- I'll do that now.

Reply:
https://github.com/pyca/cryptography/pull/8892/commits/212b75ff2f69a3b3cfc9d6a55949f23877f8f618 is my attempt at this -- it saves us the repetition of this error clause in exchange for a new error enum + `From` implementation.